flyteadmin/pkg/workflowengine/impl/prepare_execution.go

```diff
@@ -129,6 +129,9 @@ func PrepareFlyteWorkflow(data interfaces.ExecutionData, flyteWorkflow *v1alpha1
 	acceptAtWrapper := v1.NewTime(data.ExecutionParameters.AcceptedAt)
 	flyteWorkflow.AcceptedAt = &acceptAtWrapper
 
+	// Add finalizer
```

Comment:
Doesn't this already happen here?
		SetFinalizerIfEmpty(mutableW, FinalizerKey)
https://github.com/unionai/flyte/blob/14eaa16a43bff3ada4e93c6c6421c6b35471bf30/flytepropeller/pkg/controller/handler.go#L128

Reply:
@EngHabu I believe in their scenario, the CRD is getting deleted before propeller has had the chance to evaluate (and set the finalizer for) the external workflow